(MAINT) Release prep for PDK 2.1.0.0 (#1044)
puppetlabs/pdk: docs/release_notes_pdk.md

# PDK release notes

New features, enhancements, and resolved issues for the PDK 2.x release series.

## PDK 2.1

### New features and enhancements

#### Add `pdk env` subcommand

> Adds an experimental `pdk env` subcommand, which prints the current environment variables used by the PDK (https://github.com/nkanderson).

Reviewer: ```suggestion
Adds an experimental `pdk env` subcommand. Use `pdk env` to print the environment variables that PDK is currently using (https://github.com/nkanderson).
```

> Adds an experimental `pdk env` subcommand, which aids in setting a CLI context for a specified version of Puppet (https://github.com/nkanderson).

Reviewer 2: ```suggestion
Adds an experimental `pdk env` subcommand, which prints the current environment variables used by the PDK (https://github.com/nkanderson).
```

### Resolved issues

#### `--verbose` option broken for  `pdk new defined_type`

> With [ddfreyne/cri#103](https://github.com/ddfreyne/cri/issues/103) merged, this issue has been resolved.

Reviewer: ```suggestion
Fixes an issue where the `--verbose` option would not work with the `pdk new defined_type` command. See [ddfreyne/cri#103](https://github.com/ddfreyne/cri/issues/103).
```

#### PDK output scrollback in VS Code limited

> By disabling ansicon on Windows when running PDK in the integrated terminal of VS Code, this issue is now resolved.

Reviewer 2: ```suggestion
Resolved an issue using PDK in the integrated terminal of VS Code by disabling ansicon on Windows .
```

> ### Template Changes and Fixes

Reviewer: ```suggestion
### Template changes and fixes
```

Reviewer: If you could fix this on line 75 as well, that would be great :)

> * Relocation of `inventory.yaml` file used by Litmus. This prevents accidental execution of acceptance tests in production systems. (https://github.com/pmcmaw)

Reviewer: ```suggestion
* Relocated the`inventory.yaml` file used by Litmus. This prevents accidental execution of acceptance tests in production systems. (https://github.com/pmcmaw)
```

Reviewer: These template bullets have historically been in the past tense. I think that works better for these.

* Removal of win32 gems from Gemfile. These have been included in Puppet since version 5. (https://github.com/da-ar)

> * Add Puppet 7 tests to .gitlab-ci.yml (https://github.com/silug)

Reviewer: ```suggestion
* Added Puppet 7 tests to .gitlab-ci.yml (https://github.com/silug)
```
